src/AbstractOperations/computed_field.jl

```diff
@@ -75,17 +75,19 @@ end
     compute!(comp::ComputedField, time=nothing)
 
 Compute `comp.operand` and store the result in `comp.data`.
-If `time` then computation happens if `time != field.status.time`.
+If `time` then only compute dependency fields with `time != field.status.time`.
 """
 function compute!(comp::ComputedField, time=nothing)
     # First compute `dependencies`:
     compute_at!(comp.operand, time)
 
     # Now perform the primary computation
     @apply_regionally compute_computed_field!(comp)
-
     fill_halo_regions!(comp)
 
+    # Update status
+    isnothing(time) || comp.status.time = time
```

Comment:
```suggestion
    !isnothing(time) && comp.status isa FieldStatus && comp.status.time = time
```

It is possible to create fields without a status (this is a "feature" which allows people to re-use scratch fields in diagostics systems to save memory)

could also dispatch:

```julia
set_status!(status, time) = nothing
set_status!(status::FieldStatus, time::Number) = status.time = time
```